pkg/helpers/helpers.go

```diff
@@ -0,0 +1,147 @@
+package helpers
+
+import (
+	"fmt"
+
+	mcfgv1 "github.com/openshift/api/machineconfiguration/v1"
+	v1 "github.com/openshift/client-go/machineconfiguration/listers/machineconfiguration/v1"
+	ctrlcommon "github.com/openshift/machine-config-operator/pkg/controller/common"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/klog/v2"
+)
+
+const (
+	// osLabel is used to identify which type of OS the node has
+	OSLabel = "kubernetes.io/os"
+)
+
+func GetNodesForPool(mcpLister v1.MachineConfigPoolLister, nodeLister corev1listers.NodeLister, pool *mcfgv1.MachineConfigPool) ([]*corev1.Node, error) {
```

Comment:
I have a similar question as Sergio below, since both functions are implemented via controller already, did you want to consolidate them?

The functions take different inputs since the controller and operator expose things a bit differently, but it is just a bit confusing to have multiple implementations of the same code

Doesn't have to block this PR but just wanted to check if you were planning on de-duping this

Reply:
there are like 2/3 places where we don't want to update the metric. B/c it is either already 1 or already 0
so the nil spots are those places. So if the int is nil it indicates to me that we should not set the metric for no reason to the value it already is